e2e/tests/playwright-tests/helpers/operations.ts

```diff
@@ -0,0 +1,248 @@
+  page: Page,
+  userObject: { email: string, password: string, recoveryEmail: string },
+  { selectorAwaited = undefined, checkInitialValue = false } = {}
+) => {
+    await page.goto('https://gmail.com', { waitUntil: 'commit' });
+
+    if (checkInitialValue) {
+        await expect(page.locator(gmail.loginView.identifierInput))
+          .toHaveAttribute('data-initial-value', userObject.email, { timeout: 10000 });
+    } else {
+        await page.waitForSelector(gmail.loginView.identifierInput, { state: 'visible', timeout: 60000 });
+        await page.locator(gmail.loginView.identifierInput).fill(userObject.email);
+    }
+    await page.locator(gmail.loginView.identifierNextBtn).click();
+    await page.waitForSelector(gmail.loginView.passwordInput, { state: 'visible', timeout: 30000 });
+    await page.locator(gmail.loginView.passwordInput).fill(userObject.password);
+    await page.locator(gmail.loginView.passwordNextBtn).click();
+
+    const selector = selectorAwaited === undefined ? gmail.composeBtn : selectorAwaited;
+    const { recoveryEmailItem } = gmail.verifyView;
+    const { notNowBtn } = gmail.verifyView;
+    await page.waitForSelector(`:is(${selector}, ${recoveryEmailItem}, ${notNowBtn})`, { timeout: 60000 });
+    if (await page.isVisible(recoveryEmailItem)) {
+        await page.locator(recoveryEmailItem).click();
+        await page.locator(gmail.verifyView.recoveryEmailInput).fill(userObject.recoveryEmail);
+        await page.locator(gmail.verifyView.nextBtn).click();
+        await page.locator(selector).waitFor({ state: 'visible', timeout: 60000 });
+    }
+    if (await page.isVisible(notNowBtn)) await page.locator(notNowBtn).click();
+
+    await page.goto('/');
+
+    await page.waitForSelector(selectors.firstScreen.unsupportedModal.continueAnywayBtn);
+    await page.locator(selectors.firstScreen.unsupportedModal.continueAnywayBtn).click();
+
+    await page.locator(selectors.loginButton).click();
+    await page.locator(selectors.loginGoogleButton).click();
+
+    await page.waitForURL('https://local.virtru.com/');
+    await page.waitForSelector(selectors.logoutButton);
+}
+
+type EncryptDecryptProcedure = {
+    page: Page,
+    recipientEmail?: string,
+    full?: boolean
+}
+
+export const encryptDecryptProcedure = async (props: EncryptDecryptProcedure) => {
+    const { page, recipientEmail, full } = props;
+    await page.locator(selectors.draggableItem).dragTo(page.locator(selectors.dropZone));
+
+    if (recipientEmail) {
+        await page.fill('input[type=email]', recipientEmail);
+        await page.getByText(selectors.grantBtnText).click();
+        await expect(page.getByText(selectors.revokeBtnText)).toBeVisible();
+    }
+
+    await page.getByText(selectors.protectFileBtnText).click();
+
+    if (full) {
+        const responsePublicKey = await page.waitForResponse('**/auth/oidc/public-key');
+        await expect(responsePublicKey.status() === 200).toBeTruthy();
```

Comment:
Generally, reserve `toBeTruthy` for a last resort. Ideally use something that gives an informative error if it fails, in this case `toEqual` is probably the best option. See also when you decrypt a file, it would be nice to do a `toContain` on a substring.

```suggestion
        await expect(responsePublicKey.status()).toEqual(200);
```
https://playwright.dev/docs/api/class-genericassertions#generic-assertions-to-equal

Reply:
updated that and similar places